deltacat/compute/compactor/steps/materialize.py

```diff
@@ -323,8 +312,49 @@ def _materialize(compacted_tables: List[pa.Table]) -> MaterializeResult:
             write_result,
             referenced_write_result,
             np.double(peak_memory_usage_bytes),
-            np.double(emit_metrics_time),
             np.double(time.time()),
         )
 
         return merged_materialize_result
+
+
+@ray.remote
+def materialize(
+    source_partition_locator: PartitionLocator,
+    round_completion_info: Optional[RoundCompletionInfo],
+    partition: Partition,
+    mat_bucket_index: int,
+    dedupe_task_idx_and_obj_id_tuples: List[DedupeTaskIndexWithObjectId],
+    max_records_per_output_file: int,
+    compacted_file_content_type: ContentType,
+    enable_manifest_entry_copy_by_reference: bool,
+    enable_profiler: bool,
+    schema: Optional[pa.Schema] = None,
+    read_kwargs_provider: Optional[ReadKwargsProvider] = None,
+    s3_table_writer_kwargs: Optional[Dict[str, Any]] = None,
+    object_store: Optional[IObjectStore] = None,
+    deltacat_storage=unimplemented_deltacat_storage,
+    deltacat_storage_kwargs: Optional[Dict[str, Any]] = None,
+    metrics_config: Optional[MetricsConfig] = None,
+):
+    # initialize singleton on new process
+    if metrics_config:
+        MetricsConfigSingleton.instance(metrics_config)
```

Comment:
What is the benefit on this singleton wrapper class? If you are not using any methods of metrics_config nor is metrics_config compute intensive to create, what is the real benefit?

Reply:
The benefit is to create a global `metrics_config` object that cannot be accidentally overwritten.